demo/client/backend/src/fabricClient.js

```diff
@@ -0,0 +1,399 @@
+/*
+* Copyright IBM Corp All Rights Reserved
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+'use strict';
+
+const FabricCAClient = require('fabric-ca-client');
+const { Gateway, Client, FileSystemWallet, X509WalletMixin } = require('fabric-network');
+const fs = require('fs');
+const path = require('path');
+
+////////////////////////////  Global constants  /////////////////
+//  errorcodes
+const SUCCESS = 0;
+const FAILURE  = 1;
+const USER_EXISTS = 2;
+const USER_UNAUTHORIZED = 3;
+
+/////////////////////////////  input  //////////////////////////////////////////
+const orgName         = 'Org1MSP';
+const orgDomainName   = 'org1.example.com';
+const CAname          = 'ca.example.com';
+const org1Affiliation = 'org1';  //  different from MSP;  used in CA
+
+//  affiliation is used in CA;  to add to the list of affiliation, customize
+//  fabric-ca-server-config.yaml and pass in as cafiles in start command.
+//  If not, use 'org1' and 'org2' - case sensitive, which is defined in
+// the default  fabric-ca-server-config.yaml.
+
+/////////////////////////////  end of input  ///////////////////////////
+
+//  global variables loaded initially
+var configdata;
+var mspDir;
+var connectionProfile;
+var channelName;
+var chaincodeID;
+var wallet;
+var adminUserName;
+var secret;
+var bLocalHost;  //  used in setUserContext
+
+//  other global variables
+var strResponse;
+var gateway;
+var network;
+var contract;
+
+//  exported functions
+async function connectToNetwork() {
+    console.log (">>> In function, connectToNetwork");
+
+    await beginNetworkConnect();
+
+    try {
+      gateway = await new Gateway();
```

Comment:
I am wondering why this code continues when `new Gateway()` throws an error. I assume that `gateway.connect` will also fail, thus no need to continue here?!